client/src/components/RichTextEditor.js

```diff
@@ -0,0 +1,295 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import {ContentState, CompositeDecorator, Editor, EditorState, RichUtils, convertFromHTML, getDefaultKeyBinding } from 'draft-js'
+import { convertToHTML } from 'draft-convert'
+import 'draft-js/dist/Draft.css'
+import './RichTextEditor.css'
+
+class RichTextEditor extends Component {
+	static propTypes = {
+		value: PropTypes.string,
+		onChange: PropTypes.func,
+	}
+	constructor(props) {
+		super(props)
+		const decorator = new CompositeDecorator([
+			{
+				strategy: findImageEntities,
+				component: ReactImage,
+			},
+		])
+
+		this.state = {
+			editorState: EditorState.createEmpty(decorator),
+			decorator,
+			isLoaded: false,
+		}
+		this.handleOnChangeHTML = this._handleOnChangeHTML.bind(this)
```

Comment:
Use ES2015 Arrow Functions - see https://medium.com/komenco/react-autobinding-2261a1092849
```	
  handleOnChangeHTML = () => {
		const { editorState } = this.state
		const html = convertToHTML({
			entityToHTML: (entity, originalText) => {
				if (entity.type === 'IMAGE') {
					const { src, width, height, alt } = entity.data
					return <img src={imageDataSrc(src)} width={width} height={height} alt={alt} />
				}
				return originalText
			}
		})(editorState.getCurrentContent())
		this.props.onChange(html)
	}
```

Reply:
FB seems to prefer the bind method [https://reactjs.org/docs/faq-functions.html](https://reactjs.org/docs/faq-functions.html)
There are multiple blogs that contradict each other about which method is:
https://github.com/facebook/react/issues/9851
https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36

What would be the deciding factor to choose one or the other?

Reply:
good points. There seem to be multiple concerns. The last article - about not useing arrows/binds in render is probably the more worrisome - but it doesn't make a difference if we use one or the other as they are both bad in a render. I suggest to keep the use in render to a minimum, and give it a think which style to use in the general case. @gjvoosten @maradragan ?